rescompy/rescompy.py

```diff
@@ -452,37 +571,43 @@ def valid_length(
         Returns:
             valid_length (int): The computed valid length.
         """
-    
-        # Grab either the NRMSE or the RMSE.
-        if normalized:
-            errors = self.nrmse
-        else:
-            errors = self.rmse
-
-        # Calculate the valid length based on the errors and the threshold.
-        valid_length = 0
-        for error in errors:
-            if error <= threshold:
-                valid_length += 1
+        if(self.predict_length > 0):
+            # Grab either the NRMSE or the RMSE.
+            if normalized:
+                errors = self.nrmse
             else:
-                break
+                errors = self.rmse
 
-        # If the valid length is equal to the length of target_outputs, the
-        # actual valid_length might be greater than calculated; warn the user
-        # accordingly.              
-        if valid_length == self.target_outputs.shape[0]:
-            if normalized:
-                msg = "Normalized root-square-error does not exceed " \
+            # Calculate the valid length based on the errors and the threshold.
+            valid_length = 0
+            for error in errors:
+                if error <= threshold:
+                    valid_length += 1
+                else:
+                    break
+
+            # If the valid length is equal to the length of target_outputs, the
+            # actual valid_length might be greater than calculated; warn the user
+            # accordingly.              
+            if valid_length == self.target_outputs.shape[0]:
+                if normalized:
+                    msg = "Normalized root-square-error does not exceed " \
                       f" {threshold}; true valid_length may be " \
                       "greater than reported."
-            else:
-                msg = "Root-square-error does not exceed " \
+                else:
+                    msg = "Root-square-error does not exceed " \
                       f" {threshold}; true valid_length may be " \
                       "greater than reported."
-            logging.warning(msg)
+                logging.warning(msg)
+				
+            return valid_length
+
+        else:
+            msg = "valid_length is not defined for single-target mapping tasks" \
```

Comment:
should this be an error?

Reply:
Maybe a warning message and returning nan would be better?